roadmap/implementers-guide/src/runtime/scheduler.md

```diff
@@ -1,3 +1,227 @@
+- Top level responsibilities of the scheduler
+- Interfaces
+- Considerations async backing (see ticket)
+
+# Interfaces for parathreads
+
+## User Interface
+
+The entry point for the user is twofold:
+
+1. Registering a PVF with some deposit
+2. Actually bidding for a slot.
+
+### Registering a PVF with some deposit
+
+This can very likely be equal or very similar to what we have for parachains.
+
+TODO: Figure out how this works for parachains or whether there even is a
+separate deposit for the PVF. I believe parachains just have locked DOTs for
+their lease period, which also covers PVF storage.
+
+### Bidding for a slot/Auctioning System
+
+#### Goals
+
+1. Price finding
+2. Be efficient ... we are talking about slots for single blocks, the overhead
+   of assignment should be minimal.
+3. Maximize utilization: Avoid empty slots, as long as there is demand from
+   parathread willing to pay the minimum price.
+4. DoS resistance: We should not be able to get flooded with dubious bids, that
+   just fill up runtime resources.
+5. Lifetime/Update: Bids should have some lifetime or alternatively need to be
+   upgradeable. E.g. someone puts in a bid, but it is too low and is constantly
+   outbid by other bids ... eventually we either no longer care ... would like
+   to remove the bid or increase it, so it will finally make it.
+6. The user needs to know whether their bid made it through or not and when.
+
+#### General Interface
+
+The user can apply for a slot by placing a bid extrinsic. This bid will contain
+some price the user is willing to pay for a core. These extrinsic go into a
+transaction pool, as other extrinsics.
+
+There should be no reason to place bids long before we want to claim a slot. If
+spam protection mechanism are effective you should just send the bid, when you
+want to have a core in the near future ... bids will have some lifetime
+specified. If they don't manage to win a slot within that lifetime, they are
+dropped. The user would then need to bid again.
+
+Needs:
+
+1. Expose: Realistic price to pay right now. (E.g. expose recent max/lowest
+   prices/average).
+2. Actual utilization - related to price, but nevertheless good to know
+   explicitly. If there are lots of free cores right now, it is "safe" to go with
+   the minimum bid.
+3. Some form of lifetime ... Some time frame the user can expect to either get a
+   core or know for sure he won't get any. Needs to be exposed somehow? - HOW?
+4. The user does not need to be able to specify a start ... once he submits a
```

Comment:
It should be enough to allow the bids to be no earlier than N blocks into the future and not later than M blocks into the future. Get me a core asap would be just using these bounds in the extrinsic call.